common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count.py

```diff
@@ -0,0 +1,122 @@
+    """
+    This class exists to test XML import and export to/from Split.
+    """
+
+    def setUp(self):
+        super(CountMongoCallsXMLRoundtrip, self).setUp()
+        self.export_dir = mkdtemp()
+        self.addCleanup(rmtree, self.export_dir, ignore_errors=True)
+
+    @ddt.data(
+        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, 287, 780, 702, 702),
+        (MIXED_SPLIT_MODULESTORE_BUILDER, 37, 16, 190, 189),
+    )
+    @ddt.unpack
+    def test_import_export(self, store_builder, export_reads, import_reads, first_import_writes, second_import_writes):
+        with store_builder.build() as (source_content, source_store):
+            with store_builder.build() as (dest_content, dest_store):
+                source_course_key = source_store.make_course_key('a', 'course', 'course')
+                dest_course_key = dest_store.make_course_key('a', 'course', 'course')
+
+                # An extra import write occurs in the first Split import due to the mismatch between
+                # the course id and the wiki_slug in the test XML course. The course must be updated
+                # with the correct wiki_slug during import.
+                with check_mongo_calls(import_reads, first_import_writes):
+                    import_from_xml(
+                        source_store,
+                        'test_user',
+                        TEST_DATA_DIR,
+                        course_dirs=['manual-testing-complete'],
+                        static_content_store=source_content,
+                        target_course_id=source_course_key,
+                        create_course_if_not_present=True,
+                        raise_on_failure=True,
+                    )
+
+                with check_mongo_calls(export_reads):
+                    export_to_xml(
+                        source_store,
+                        source_content,
+                        source_course_key,
+                        self.export_dir,
+                        'exported_source_course',
+                    )
+
+                with check_mongo_calls(import_reads, second_import_writes):
+                    import_from_xml(
+                        dest_store,
+                        'test_user',
+                        self.export_dir,
+                        course_dirs=['exported_source_course'],
+                        static_content_store=dest_content,
+                        target_course_id=dest_course_key,
+                        create_course_if_not_present=True,
+                        raise_on_failure=True,
+                    )
+
+
+@ddt.ddt
+class CountMongoCallsCourseTraversal(TestCase):
+    """
+    Tests the number of Mongo calls made when traversing a course tree from the top course root
+    to the leaf nodes.
+    """
+
+    @ddt.data(
+        (None, 7),  # The way this traversal *should* be done.
+        (0, 145)    # The pathological case - do *not* query a course this way!
+    )
```

Comment:
You got it - read the numbers and weep.

On Thu, Feb 12, 2015 at 3:31 PM, Calen Pennington <notifications@github.com>
wrote:

> In common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count.py
> <https://github.com/edx/edx-platform/pull/6920#discussion_r24617292>:
>
> > +                        create_course_if_not_present=True,
> > +                        raise_on_failure=True,
> > +                    )
> > +
> > +
> > +@ddt.ddt
> > +class CountMongoCallsCourseTraversal(TestCase):
> > +    """
> > +    Tests the number of Mongo calls made when traversing a course tree from the top course root
> > +    to the leaf nodes.
> > +    """
> > +
> > +    @ddt.data(
> > +        (None, 7),  # The way this traversal *should* be done.
> > +        (0, 145)    # The pathological case - do *not* query a course this way!
> > +    )
>
> Is it worth testing this w/ old-mongo as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/6920/files#r24617292>.
>